src/main/java/com/aws/greengrass/security/SecurityService.java

```diff
@@ -241,7 +242,7 @@ public static URI uriFromPossibleFileURIString(String path) {
         }
 
         @Override
-        public KeyPair getKeyPair(URI privateKeyUri)
+        public KeyPair getKeyPair(URI privateKeyUri, URI certificateUri)
```

Comment:
Shouldn't it use public key from certificate instead of generating public key from private key?

Reply:
No, why bother? When it is file based we're able to generate a public key from private key, that generated key is guaranteed to be correct. We do not need the certificate in this case at all, but it is part of the interface for PKCS11 where generating public from private isn't feasible.

Reply:
Fair, for this use case.